### DISCUSSION TITLE
NetBox lines of Python code

### DISCUSSION THREAD

**Discussion by markkuleinio** [netbox-community/netbox#11199] [General]
To whoever is interested 😁

```
(venv) markku@devel:~/temp/netbox$ git pull
...
From https://github.com/netbox-community/netbox
...
Already up to date.
(venv) markku@devel:~/temp/netbox$ pip install lopc
Collecting lopc
  Using cached lopc-1.0.2-py3-none-any.whl (4.6 kB)
Installing collected packages: lopc
Successfully installed lopc-1.0.2
(venv) markku@devel:~/temp/netbox$ pip show lopc | grep Summary
Summary: Counts lines of Python code
(venv) markku@devel:~/temp/netbox$ lopc .
. Files: 666 Lines: 89709
(venv) markku@devel:~/temp/netbox$
```

**Comment by jeremystretch** [comment]
Neat, thanks for sharing. I wonder if it's work tracking "lines of code" as a metric over time.

**Comment by zenion** [comment]
"Measuring programming progress by lines of code is like measuring aircraft building progress by weight"
-- Bill Gates

🤣 

but seriously, would be a fun metric to watch on the repo haha

**Comment by abhi1693** [comment]
Try running cloc because this repo also consists of HTML, ts, css, md etc. I believe that will bring the final count somewhere in the neighborhood of 120k lines

**Comment by mtinberg** [comment]
I was inspired to fiddle with this a bit in PowerShell (instead of bash / xargs)

foreach ($tag in @(git tag --list) { perl ../cloc/cloc --vcs=git  --json --report-file="../cloc-$tag.json"  --git $tag; }

trying to generate the history so it can be extracted and popped into gnuplot or Excel for visualization, if you are trying to see how the project changes over time.  It was a bit slow to run for every version, and I tried to figure out how to parallelize it using Start-Job but didn't quite figure it out in the time I had to mess around with it.  I didn't get to extracting the data into tabular form, maybe next time 😉


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
